drivers/pse/si3474/si3474.c

```diff
@@ -0,0 +1,383 @@
+						  &events[POWER_EVENT]);
+			if (res != 0) {
+				LOG_WRN("Si3474 reading event status table failed");
+			}
+			break;
+		case DISCONNECT_IT:
+			res = si3474_i2c_read_reg(cfg->i2c.bus, dev_address,
+						  DISCONNECT_PCUT_FAULT_REG,
+						  &events[DISCONNECT_PCUT_FAULT]);
+			if (res != 0) {
+				LOG_WRN("Si3474 reading event status table failed");
+			}
+			res = si3474_i2c_read_reg(cfg->i2c.bus, dev_address, SUPPLY_EVENT_REG,
+						  &events[SUPPLY_EVENT]);
+			if (res != 0) {
+				LOG_WRN("Si3474 reading event status table failed");
+			}
+			break;
+
+		case DETECT_CC_DONE_IT:
+			mask <<= 1;
+			cnt++;
+		case CLASS_DONE_IT:
+			res = si3474_i2c_read_reg(cfg->i2c.bus, dev_address, CLASS_DETECT_EVENT_REG,
+						  &events[CLASS_DETECT_EVENT]);
+			if (res != 0) {
+				LOG_WRN("Si3474 reading event status table failed");
+			}
+			break;
+
+		case P_I_FAULT_IT:
+			res = si3474_i2c_read_reg(cfg->i2c.bus, dev_address,
+						  DISCONNECT_PCUT_FAULT_REG,
+						  &events[DISCONNECT_PCUT_FAULT]);
+			if (res != 0) {
+				LOG_WRN("Si3474 reading event status table failed");
+			}
+			res = si3474_i2c_read_reg(cfg->i2c.bus, dev_address, ILIM_START_FAULT_REG,
+						  &events[ILIM_START_FAULT]);
+			if (res != 0) {
+				LOG_WRN("Si3474 reading event status table failed");
+			}
+			res = si3474_i2c_read_reg(cfg->i2c.bus, dev_address, SUPPLY_EVENT_REG,
+						  &events[SUPPLY_EVENT]);
+			if (res != 0) {
+				LOG_WRN("Si3474 reading event status table failed");
+			}
+			break;
+
+		case START_EVENT_IT:
+			res = si3474_i2c_read_reg(cfg->i2c.bus, dev_address, ILIM_START_FAULT_REG,
+						  &events[ILIM_START_FAULT]);
+			if (res != 0) {
+				LOG_WRN("Si3474 reading event status table failed");
+			}
+			res = si3474_i2c_read_reg(cfg->i2c.bus, dev_address, POWER_ON_FAULT_REG,
+						  &events[POWER_ON_FAULT]);
+			if (res != 0) {
+				LOG_WRN("Si3474 reading event status table failed");
+			}
+			break;
+
+		case SUPPLY_EVENT_IT:
+			res = si3474_i2c_read_reg(cfg->i2c.bus, dev_address, SUPPLY_EVENT_REG,
+						  &events[SUPPLY_EVENT]);
+			if (res != 0) {
+				LOG_WRN("Si3474 reading event status table failed");
+			}
+			break;
+		}
+		mask <<= 1;
+	}
+	return res;
+}
+
+static int si3474_set_events(const struct device *dev, uint8_t events)
+{
+	const struct si3474_config *cfg = dev->config;
+	uint16_t dev_address = cfg->i2c.addr;
+	int res;
+
+	res = i2c_reg_write_byte(cfg->i2c.bus, dev_address, INTERRUPT_MASK, events);
+	if (res != 0) {
+		LOG_WRN("Si3474 setting events failed!");
+		return res;
+	}
+	return res;
+}
+
+static int si3474_get_current(const struct device *dev, uint8_t channel, uint16_t *val)
+{
+	const struct si3474_config *cfg = dev->config;
+	uint16_t dev_address = cfg->i2c.addr;
+	uint8_t buff[CURRENT_SIZE_BUFFER];
+	int res = 0;
+
+	switch (channel) {
+	case CHANNEL_0:
+		res = i2c_burst_read(cfg->i2c.bus, dev_address, PORT1_CURRENT_LSB_REG, &buff[0],
```

Comment:
This is the only line that seem to change from each case. Is it possible to store the define in a variable and move the redundant code outside the switch case?
The voltage function seem to do pretty much the same. Maybe write a helper function.

Reply:
Yes, correct, the redundant code can be moved outside of the switch to reduce the source code. This will be fixed.